dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -91,6 +92,18 @@ public KernelBuilder WithMemoryStorage(IMemoryStore storage)
         return this;
     }
 
+    /// <summary>
+    /// Add prompt template engine to the kernel to be built.
+    /// </summary>
+    /// <param name="promptTemplateEngine">Prompt template engine to add.</param>
+    /// <returns>Updated kernel builder including the prompt template engine.</returns>
+    public KernelBuilder WithPromptTemplateEngine(IPromptTemplateEngine promptTemplateEngine)
```

Comment:
Can this all be just: AddAIService<IPromptTemplateEngine>(promptTemplateEngine); ?

Reply:
You probably meant AddService<IPromptTemplateEngine>(promptTemplateEngine) instead of Add**AI**Service<IPromptTemplateEngine>(promptTemplateEngine), right?

If that is the case, then the answer is yes, we can remove the WithPromptTemplateEngine, WithLogger, and WithMemory methods, but the question we should ask ourselves is whether we should do it. Removing these methods would make it difficult for developers to understand the capabilities of the builder. Without these methods, when developers type **new KernelBuilder().**, the only method that Intellisense will show them is AddService<T>, which is not very informative. On the other hand, with all the With* methods shown by Intellisense, developers can immediately start exploring them in the IDE by looking at their signatures to familiarize themselves with the public API surface of the class. Although this is not a replacement for official documentation, it is a good addition to it.